scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala

```diff
@@ -76,7 +76,42 @@ class GitOpsImpl(private[util] val workingDirectory: AbsoluteFile)
     )
     for {
       root <- rootDir.toSeq
-      path <- exec(cmd).toOption.toSeq.flatten
+      path <- exec(cmd).getOrElse(Seq.empty)
     } yield AbsoluteFile.fromFile(new File(path), root)
   }
+
+  override def status: Seq[AbsoluteFile] = {
+    val cmd = Seq(
+      "git",
+      "status",
+      "--porcelain"
+    )
+    for {
+      root <- rootDir.toSeq
+      statusLine <- exec(cmd).getOrElse(Seq.empty)
+      file <- getFileFromGitStatusLine(statusLine).toOption.toSeq
+    } yield AbsoluteFile.fromFile(file, root)
+  }.filter(_.exists)
+
+  private final val renameStatusCode = "R"
+  private final val renameStatusArrowDelimiter = "-> "
+
+  private def extractPathPart(s: String): Try[String] =
+    Try(
+      Option(s)
+        .filter(_.substring(0, 2).contains(renameStatusCode))
+        .map(_.split(renameStatusArrowDelimiter).last)
+        .getOrElse(s.trim.split(' ').tail.mkString(" "))
+        .trim
+    )
+
+  private def trimQuotes(s: String): String =
+    s.replaceAll("^\"|\"$", "")
+
+  private def getFileFromGitStatusLine(s: String): Try[File] =
+    for {
+      pathRaw <- extractPathPart(s)
+      path = trimQuotes(pathRaw)
+      file <- Try(new File(path))
+    } yield file
```

Comment:
Can we remove the `Try` block from `Try(new File(path))` and change it to following? I think we don't have to catch the exceptions for `new File` because [`File` constructor throws exception only when `path` is null](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#File-java.lang.String-)

```scala
  // ...
  path = trimQuotes(pathRaw)
} yield new File(path)
```